js/cards.js

```diff
@@ -114,6 +152,56 @@ export async function defineCardComponent() {
   customElements.define("tcg-card", Card);
 }
 
+function addAnimation(e){
+  // apply the animated class to start the CSS animation
+  e.target.classList.add("animated");
+}
+function updateZIndex(e){
+    e.target.style.zIndex = card_z_index;
+    // we don't want this to update again when reanimating later
+    e.target.removeEventListener("animationstart", updateZIndex);
+}
+
+function setupHover(){
+  // setup the initial event listeners for all cards
+  let cardList = document.getElementsByClassName("card-image");
+  for (let i = 0; i < cardList.length; i++) {
+    cardList[i].addEventListener("animationend", function(){
+      cardList[i].classList.remove("animated");
+    })
+    cardList[i].addEventListener("mouseover", addAnimation);
+    cardList[i].addEventListener("click", addAnimation);
+  }
+}
+export function addHover(){
+  // remove any unclickable classes from all the opened cards, so all can be interacted with
+  let clickedCard = document.getElementsByClassName("opened");
+  for (let i = 0; i < clickedCard.length; i++) {
+    clickedCard[i].classList.remove("unclickable");
+  }
+}
+export function removeHover(){
```

Comment:
`onlyHoverTopmostCard` or `transitionToPileHover` would be a clearer name (though a mouthful)

And to double check my understanding, this runs once on transition from grid to pile. Then, `flipCard` handles subsequent hover removal on the previously flipped card, and adds it to the one that was just flipped? Might be good to comment that

Reply:
Oh whoops you mentioned it here too. Updated it again and extra comment added. I don't really care if the name is long, if it makes sense to you. (It made sense to me but of course not for everyone. :))